### PR TITLE
[Documentation] Show faster build command first in docs/development.md

### PR DESCRIPTION
**Pull request by renxida**

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -62,33 +64,40 @@ cmake -GNinja -Bbuild \
...
-```
-* Enabling libtorch binary cache
-By default we download the latest version of libtorch everytime you build so we are always on the latest version. Set `-DLIBTORCH_CACHE=ON` to
-not download the latest version everytime. If libtorch gets out of date and you test against a newer PyTorch you may notice failures.
-```shell
-  -DLIBTORCH_CACHE=ON
-```
-* Enabling building libtorch as part of your build
-By default we download the latest version of libtorch. We have an experimental path to build libtorch (and PyTorch wheels) from source.
+  externals/llvm-project/llvm \
```
Comment (kuhar) [inline docs/development.md:67]: I'd move the source dir to the top

File: docs/development.md [hunk 2]
```diff
@@ -62,33 +64,40 @@ cmake -GNinja -Bbuild \
...
+  `# use clang`\
```
Comment (kuhar) [inline docs/development.md:68]: Why do we need backticks here?

File: docs/development.md [hunk 3]
```diff
@@ -62,33 +64,40 @@ cmake -GNinja -Bbuild \
...
+  -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ \
+  `# use ccache to cache build results` \
+  -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+  `# use LLD to link in seconds, rather than minutes` \
+  `# if using clang <= 13, replace --ld-path=lld with -fuse-ld=lld` \
+  -DCMAKE_EXE_LINKER_FLAGS_INIT="--ld-path=lld" \
+  -DCMAKE_MODULE_LINKER_FLAGS_INIT="--ld-path=lld" \
+  -DCMAKE_SHARED_LINKER_FLAGS_INIT="--ld-path=lld" \
```
Comment (kuhar) [inline docs/development.md:76]: Does this work? I thought you need `ld.lld`

File: docs/development.md [hunk 4]
```diff
@@ -62,33 +64,40 @@ cmake -GNinja -Bbuild \
...
+  `# Enabling libtorch binary cache instead of downloading the latest libtorch everytime.` \
+  `# Testing against a mismatched version of libtorch may cause failures` \
+  -DLIBTORCH_CACHE=ON \
+  `# Enable an experimental path to build libtorch (and PyTorch wheels) from source,` \
```
Comment (kuhar) [inline docs/development.md:80]: This one is misaligned

File: docs/development.md [hunk 5]
```diff
@@ -53,6 +53,8 @@ Two setups are possible to build: in-tree and out-of-tree. The in-tree setup is
 
 The following command generates configuration files to build the project *in-tree*, that is, using llvm/llvm-project as the main build. This will build LLVM as well as torch-mlir and its subprojects.  On Windows, use the "Developer PowerShell for Visual Studio" to ensure that the compiler and linker binaries are in the `PATH` variable.
 
+This requires `lld`, `clang`, `ccache`, and other dependencies for building `libtorch` / `PyTorch` wheels from source. If you run into issues because of these, try the [simplified build command](#simplified-build).
+
 ```shell
 cmake -GNinja -Bbuild \
   -DCMAKE_BUILD_TYPE=Release \
```
Comment (kuhar) [inline docs/development.md:60]: I'd also recommend enabling assertions

**Review by kuhar — CHANGES_REQUESTED** [review]

File: docs/development.md (hunk 2 above)
Comment (renxida) [reply docs/development.md:68]: newline escapes don't work inside comments

Comment (renxida) [reply docs/development.md:68]: it's a trick to put comments into multi line commands

File: docs/development.md (hunk 3 above)
Comment (renxida) [reply docs/development.md:76]: worked for me! ubuntu 22.04

File: docs/development.md (hunk 4 above)
Comment (renxida) [reply docs/development.md:81]: couldn't find the misalignment. could you check if it's still there?

File: docs/development.md (hunk 5 above)
Comment (renxida) [reply docs/development.md:61]: added

File: docs/development.md (hunk 3 above)
Comment (kuhar) [reply docs/development.md:77]: Maybe there's a special case that detects lld but not lld-version? In any case, I'm fine with this if it works.

**Review by kuhar — APPROVED** [review]
Thanks, this is very useful

File: docs/development.md [hunk 6]
```diff
@@ -53,42 +53,52 @@ Two setups are possible to build: in-tree and out-of-tree. The in-tree setup is
 
 The following command generates configuration files to build the project *in-tree*, that is, using llvm/llvm-project as the main build. This will build LLVM as well as torch-mlir and its subprojects.  On Windows, use the "Developer PowerShell for Visual Studio" to ensure that the compiler and linker binaries are in the `PATH` variable.
 
+This requires `lld`, `clang`, `ccache`, and other dependencies for building `libtorch` / `PyTorch` wheels from source. If you run into issues because of these, try the [simplified build command](#simplified-build).
+
 ```shell
 cmake -GNinja -Bbuild \
+  externals/llvm-project/llvm \
   -DCMAKE_BUILD_TYPE=Release \
+  -DLLVM_ENABLE_ASSERTIONS=ON" \
```
Comment (kuhar) [inline docs/development.md:62]: Trailing quote?

Comment (renxida) [reply docs/development.md:62]: ack ty

**Pull request merged**